Objects/longobject.c

```diff
@@ -669,10 +669,18 @@ PyLong_AsLong(PyObject *obj)
     int overflow;
     long result = PyLong_AsLongAndOverflow(obj, &overflow);
     if (overflow) {
-        /* XXX: could be cute and give a different
-           message for overflow == -1 */
-        PyErr_SetString(PyExc_OverflowError,
-                        "Python int too large to convert to C long");
+        if (overflow > 0) {
+            PyErr_Format(PyExc_OverflowError,
+                         "Python int too large to convert to C long: "
+                         "value would be greater than %ld",
+                         LONG_MAX);
+        }
+        else {
+            PyErr_Format(PyExc_OverflowError,
+                         "Python int too large to convert to C long: "
```

Comment:
Maybe 'too small' reads better here (and below)?

Reply:
Could you be more precise?  Replacing "too large" with "too small" will be just plain wrong right here.

Reply:
We are in the negative case here and "too large" combined with "less than" rings a bell for me. Maybe just splitting hairs (or I am completely mistaken).

Reply:
My english is bad, so I trust your judgement.

Though, I don't see a problem so far.  This exception supposed to
mean: value too large (by magnitude) for long, it would be < LONG_MIN.


```suggestion
                         "Python int magnitude too large to convert to C long: "
```
?

Reply:
See Serhiy's comment below. I agree it would be better to just have simpler and unified messages.